web/src/settings_components.ts

```diff
@@ -1013,6 +1027,20 @@ export function populate_data_for_realm_settings_request(
                     });
                     continue;
                 }
+                if (property_name === "direct_message_initiator_group") {
+                    data[property_name] = JSON.stringify({
+                        new: input_value,
+                        old: realm.realm_direct_message_initiator_group,
+                    });
+                    continue;
+                }
+                if (property_name === "direct_message_permission_group") {
+                    data[property_name] = JSON.stringify({
+                        new: input_value,
+                        old: realm.realm_direct_message_permission_group,
+                    });
+                    continue;
```

Comment:
I think we should just handle all group settings together either by passing a variable like `for_api_value` in `get_input_element_value`, rather than having `if` statements for each setting.

Would be best to do that as a follow-up. Will try if I can manage something which is not much hard to read.

Reply:
I updated the code to just add the setting to a set and just use a single if-block.